src/libraries/System.Private.CoreLib/src/System/Collections/Generic/ArraySortHelper.cs

```diff
@@ -556,45 +556,44 @@ private static void InsertionSort(Span<T> keys)
         // - The floating-point comparisons here assume no NaNs, which is valid only because the sorting routines
         //   themselves special-case NaN with a pre-pass that ensures none are present in the values being sorted
         //   by moving them all to the front first and then sorting the rest.
-        // - The `? true : false` is to work-around poor codegen: https://github.com/dotnet/runtime/issues/37904#issuecomment-644180265.
         // - These are duplicated here rather than being on a helper type due to current limitations around generic inlining.
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)] // compiles to a single comparison or method call
         private static bool LessThan(ref T left, ref T right)
         {
-            if (typeof(T) == typeof(byte)) return (byte)(object)left < (byte)(object)right ? true : false;
-            if (typeof(T) == typeof(sbyte)) return (sbyte)(object)left < (sbyte)(object)right ? true : false;
-            if (typeof(T) == typeof(ushort)) return (ushort)(object)left < (ushort)(object)right ? true : false;
-            if (typeof(T) == typeof(short)) return (short)(object)left < (short)(object)right ? true : false;
-            if (typeof(T) == typeof(uint)) return (uint)(object)left < (uint)(object)right ? true : false;
-            if (typeof(T) == typeof(int)) return (int)(object)left < (int)(object)right ? true : false;
-            if (typeof(T) == typeof(ulong)) return (ulong)(object)left < (ulong)(object)right ? true : false;
-            if (typeof(T) == typeof(long)) return (long)(object)left < (long)(object)right ? true : false;
-            if (typeof(T) == typeof(nuint)) return (nuint)(object)left < (nuint)(object)right ? true : false;
-            if (typeof(T) == typeof(nint)) return (nint)(object)left < (nint)(object)right ? true : false;
-            if (typeof(T) == typeof(float)) return (float)(object)left < (float)(object)right ? true : false;
-            if (typeof(T) == typeof(double)) return (double)(object)left < (double)(object)right ? true : false;
-            if (typeof(T) == typeof(Half)) return (Half)(object)left < (Half)(object)right ? true : false;
-            return left.CompareTo(right) < 0 ? true : false;
+            if (typeof(T) == typeof(byte)) return (byte)(object)left < (byte)(object)right;
```

Comment:
why these if’s and not using switch expressions for these in this file?

Reply:
The IL patterns of `typeof(T) == constant` and `(the same type)(object)value` are specially recognized by JIT and will be reduced to a nop. Other patterns are harder to analysis.

Reply:
Also, `typeof()` isn't a constant value and can't be used in a switch statement/expression.